botnim/document_parser/dynamic_extractions/pdf_extraction/pdf_pipeline.py

```diff
@@ -0,0 +1,461 @@
+                logger.error(error_message)
+                return None
+            
+            # Step 3: Prepare output data for each entity
+            results = []
+            for i, extracted_fields in enumerate(extracted_fields_list):
+                # Prepare metadata
+                metadata = self._prepare_metadata(source_config, pdf_path, i, len(extracted_fields_list))
+                
+                # Create result object
+                result = self._create_result_object(extracted_fields, metadata, source_config, pdf_path, i)
+                results.append(result)
+                
+                # Upload to Google Sheets immediately if enabled
+                self._upload_entity_to_sheets(result, source_config, sheets_sync, spreadsheet_id, sheet_name, replace_sheet, i)
+            
+            total_time = time.time() - start_time
+            logger.info(f"Successfully processed PDF: {pdf_path} - extracted {len(results)} entities")
+            
+            # Record metrics
+            if self.metrics:
+                metrics = ExtractionMetrics(
+                    pdf_path=pdf_path,
+                    source_name=source_config.name,
+                    text_extraction_time=text_extraction_time,
+                    field_extraction_time=field_extraction_time,
+                    total_processing_time=total_time,
+                    text_length=len(text),
+                    entities_extracted=len(results),
+                    success=True
+                )
+                self.metrics.record_extraction(metrics)
+            
+            return results
+            
+        except Exception as e:
+            total_time = time.time() - start_time
+            error_message = str(e)
+            logger.error(f"Failed to process PDF {pdf_path}: {e}")
+            
+            # Record metrics for failed extraction
+            if self.metrics:
+                metrics = ExtractionMetrics(
+                    pdf_path=pdf_path,
+                    source_name=source_config.name,
+                    text_extraction_time=text_extraction_time,
+                    field_extraction_time=field_extraction_time,
+                    total_processing_time=total_time,
+                    text_length=0,
+                    entities_extracted=0,
+                    success=False,
+                    error_message=error_message
+                )
+                self.metrics.record_extraction(metrics)
+            
+            return None
+    
+    def find_pdf_files(self, source_config) -> List[str]:
+        """
+        Find PDF files matching the source pattern.
+        
+        Args:
+            source_config: Source configuration from YAML
+            
+        Returns:
+            List of PDF file paths
+        """
+        pattern = source_config.file_pattern
+        pdf_files = []
+        
+        try:
+            # Handle relative paths by resolving from config file location
```

Comment:
Please lets have a single exact unambiguous interpretation of what the path should be in the config without all these fallbacks which will only result in us parsing and indexing the wrong thing.